### PR TITLE
hwmv2: ite_ec: port to HWMv2

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Port ite_ec soc and boards to hwmv2.

### CODE REVIEW

File: soc/ite/ec/Kconfig.defconfig
Comment (nordicjm) [inline soc/ite/ec/Kconfig.defconfig:4]: `rsource`

File: soc/ite/ec/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/Kconfig.soc:4]: `rsource`

File: soc/ite/ec/it8xxx2/Kconfig.defconfig.series
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.defconfig.series:57]: `rsource`

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:13]: `SOC_FAMILY_ITE_EC` stays in `Kconfig.soc` file, the rest go in `Kconfig`, choice selection needs to be removed

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:17]: Prompts vanish and the below move to `Kconfg` as per above, the part that says in this file is minimal, basically:
```
config SOC_IT8XXX2
    bool
    select SOC_SERIES_ITE_IT8XXX2
```

File: soc/ite/ec/Kconfig
Comment (nordicjm) [inline soc/ite/ec/Kconfig:15]: And this would want putting back as `rsource` for `Kconfig`

File: boards/ite/it8xxx2_evb/Kconfig.it8xxx2_evb
Comment (nordicjm) [inline boards/ite/it8xxx2_evb/Kconfig.it8xxx2_evb:5]:
```suggestion
```

**Review by gmarull — CHANGES_REQUESTED** [review]
update docs/maintainers as needed

**Comment by fabiobaltieri** [comment]
Compliance is still not having it though, is there some known issues with Kconfig symbols?

File: soc/ite/ec/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/Kconfig.soc:8]:
```suggestion
```

File: soc/ite/ec/it8xxx2/Kconfig
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig:6]: `	select SOC_FAMILY_ITE_EC` should be in `Kconfig.soc`

File: soc/ite/ec/it8xxx2/Kconfig
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig:22]: Were these in the original board file? Seems they were, that is odd

File: boards/ite/it82xx2_evb/Kconfig.defconfig
Comment (nordicjm) [inline boards/ite/it82xx2_evb/Kconfig.defconfig:4]: Needs an index.rst in the ite folder, copy one of the other ones and update vendor name and title

Comment (fabiobaltieri) [reply boards/ite/it82xx2_evb/Kconfig.defconfig:4]: done

File: soc/ite/ec/it8xxx2/Kconfig
Comment (nordicjm) [reply soc/ite/ec/it8xxx2/Kconfig:21]: Let's remove these and `HAS_FLASH_LOAD_OFFSET` since they should not be required and flash load offset is 0

File: soc/ite/ec/Kconfig.soc
Comment (fabiobaltieri) [reply soc/ite/ec/Kconfig.soc]: done

File: soc/ite/ec/it8xxx2/Kconfig
Comment (fabiobaltieri) [reply soc/ite/ec/it8xxx2/Kconfig]: ok done

File: soc/ite/ec/it8xxx2/Kconfig
Comment (fabiobaltieri) [reply soc/ite/ec/it8xxx2/Kconfig:21]: interesting, seems like this should be moved in `drivers/flash/Kconfig.it8xxx2` but maybe it's better to leave it here for now so these patches are confined under `soc/`

Comment (nordicjm) [reply soc/ite/ec/it8xxx2/Kconfig:21]: Removed, not moved :D there is no reason the flash driver must be enabled for this board that I can see

Comment (fabiobaltieri) [reply soc/ite/ec/it8xxx2/Kconfig]: okay:

```
        select FLASH
        select FLASH_HAS_PAGE_LAYOUT
        select FLASH_HAS_DRIVER_ENABLED
        select HAS_FLASH_LOAD_OFFSET
```

all gone

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:14]: These can go to `Kconfig`

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:29]: `select SOC_IT8XXX2_REG_SET_V1` to `Kconfig`

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:50]: These 2 to `Kconfig` (and for below instances too)

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:64]: this line to new line number 157 to `Kconfig`

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (nordicjm) [inline soc/ite/ec/it8xxx2/Kconfig.soc:66]: can add `select FLASH` to this, seems it is needed but only for this

**Review by nordicjm — COMMENTED** [review]
minor changes, should be good to go after this

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (fabiobaltieri) [reply soc/ite/ec/it8xxx2/Kconfig.soc]: done

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (fabiobaltieri) [reply soc/ite/ec/it8xxx2/Kconfig.soc]: done, moved all this stuff

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (fabiobaltieri) [reply soc/ite/ec/it8xxx2/Kconfig.soc]: cool

File: soc/ite/ec/it8xxx2/Kconfig.soc
Comment (fabiobaltieri) [reply soc/ite/ec/it8xxx2/Kconfig.soc]: done

**Comment by fabiobaltieri** [comment]
@nordicjm thanks, all done, compliance seems to be happy now even though I don't quite understand why, guess the behavior of the v2 Kconfig check

**Review by nordicjm — APPROVED** [review]

**Comment by nordicjm** [comment]
> @nordicjm thanks, all done, compliance seems to be happy now even though I don't quite understand why, guess the behavior of the v2 Kconfig check

`Kconfig.soc` is used by things like documentation whereas `Kconfig` is only used by the actual image build itself

**Review by gmarull — APPROVED** [review]

**Pull request merged**
